libs/scheduler/aws.go

```diff
@@ -45,6 +45,10 @@ func (job *Job) PopulateAwsCredentialsEnvVarsForJob() error {
 		log.Printf("Project-level AWS role detected, Assuming role for project: %v", job.ProjectName)
 		var err error
 		backendConfigArgs, err := populateretrieveBackendConfigArgs(*job.StateEnvProvider)
+
+		// Terragrunt will cause a backend configuration problem if backend-config options are passed and envs of the same key are passed. 
+		// which will trigger a request to init with --reconfigure, so do not use backend-config for terragrunt	
+		if job.Terragrunt != true {
```

Comment:
today I learned something new, so with terragrunt does it pass those options to terraform based on terragrunt.hcl configuration or what is the source of the duplicates?

Reply:
From the docs I read it should not happen but in practice I tested it out, if I set aws credentials in the my environment and then also pass in backend-config options for credentials causes that request on state change when it runs the plan. 

The `init` would work fine with both backend-config and environment variables
then when the plan would run since only backend-config variables were passed into `init` when the plan would run using environment variables it would detect a remote change the backend-config vars are not passed into plan. 

So the option was to make it pass backend-config to all steps or just exclude it.  I chose to exclude it path, as it does not seem to impact terraform. backend-config is not recommended on the TF site, I figured it was best at least for TG to just ignore their use. 

I did not go much deeper than that, but I know the above does work.